File: CS/EFDataSourceSnippets/Code.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.DataAccess.EntityFramework;

namespace EFDataSourceSnippets
{
    public class Test
    {
        EFDataSource DataSource { get; set; }
        public Test()
        {
            DataSource = InitializeEFDataSource();
        }
        EFDataSource InitializeEFDataSource()
        {
            return GenerateEFDataSource();
        }
        const string connectionString = "Data Source=localhost;Initial Catalog=Northwind;Persist Security Info=True;User ID=sa;Password=pwd;MultipleActiveResultSets=True;Application Name=EntityFramework";
        private EFDataSource GenerateEFDataSource()
        {
            var connectionParameters = new DevExpress.DataAccess.EntityFramework.EFConnectionParameters(typeof(NorthwindEntities), "EntityConnection", connectionString);
            var efDataSource = new DevExpress.DataAccess.EntityFramework.EFDataSource(connectionParameters);

            ((System.ComponentModel.ISupportInitialize)(efDataSource)).BeginInit();
            efDataSource.StoredProcedures.Add(GetStoredProcedure());
            ((System.ComponentModel.ISupportInitialize)(efDataSource)).EndInit();
            return efDataSource;
        }
        private EFStoredProcedureInfo GetStoredProcedure()
        {
            var efParameter1 = new DevExpress.DataAccess.EntityFramework.EFParameter("CustomerID", typeof(string), "ALFKI");
            var efStoredProc = new DevExpress.DataAccess.EntityFramework.EFStoredProcedureInfo("CustOrderHist", new[] { efParameter1 });
            return efStoredProc;
        }
    }
}

File: VB/EFDataSourceSnippets/Code.vb
Imports DevExpress.DataAccess.EntityFramework

Namespace EFDataSourceSnippets

    Public Class Test

        Private Property DataSource As EFDataSource

        Public Sub New()
            DataSource = InitializeEFDataSource()
        End Sub

        Private Function InitializeEFDataSource() As EFDataSource
            Return GenerateEFDataSource()
        End Function

        Const connectionString As String = "Data Source=localhost;Initial Catalog=Northwind;Persist Security Info=True;User ID=sa;Password=pwd;MultipleActiveResultSets=True;Application Name=EntityFramework"

        Private Function GenerateEFDataSource() As EFDataSource
            Dim connectionParameters = New EFConnectionParameters(GetType(NorthwindEntities), "EntityConnection", connectionString)
            Dim efDataSource = New EFDataSource(connectionParameters)
            CType(efDataSource, System.ComponentModel.ISupportInitialize).BeginInit()
            efDataSource.StoredProcedures.Add(GetStoredProcedure())
            CType(efDataSource, System.ComponentModel.ISupportInitialize).EndInit()
            Return efDataSource
        End Function

        Private Function GetStoredProcedure() As EFStoredProcedureInfo
            Dim efParameter1 = New EFParameter("CustomerID", GetType(String), "ALFKI")
            Dim efStoredProc = New EFStoredProcedureInfo("CustOrderHist", {efParameter1})
            Return efStoredProc
        End Function
    End Class
End Namespace
